src/contexts/productsContext.js

```diff
@@ -0,0 +1,39 @@
+import axios from "axios";
+import { createContext, useContext, useState, useEffect } from "react";
+
+const ProductContext = createContext(null);
+
+const useProducts = () => useContext(ProductContext);
+
+const ProductsProvider = ({ children }) => {
+  const [products, setProducts] = useState([]);
+  const [categories, setCategoies] = useState([]);
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await axios.get("/api/products");
+        const data = response.data.products;
+        setProducts(data);
+        setCategoies(
+          data.reduce(
+            (acc, curr) =>
+              acc.includes(curr.categoryName)
+                ? acc
+                : [...acc, curr.categoryName],
+            []
```

Comment:
Nice implementation for fetching categories

Reply:
just a suggestion we can create different functions to get API calls so that we can reuse